lewis/adapters/stream.py

```diff
@@ -128,18 +162,71 @@ def __init__(self, target_method, regex, regex_flags=0, argument_mappings=None,
         self.return_mapping = return_mapping
         self.doc = doc
 
-    def map_arguments(self, arguments):
-        """
-        Returns the mapped function arguments. If no mapping functions are defined, the arguments
-        are returned as they were supplied.
+    @property
+    def patterns(self):
+        return [self.pattern.pattern]
 
-        :param arguments: List of arguments for bound function as strings.
-        :return: Mapped arguments.
-        """
-        if not self.argument_mappings:
-            return arguments
+    def process_request(self, request, target):
+        match = self.pattern.match(request)
 
-        return [f(a) for f, a in zip(self.argument_mappings, arguments)]
+        if not match:
+            raise CanNotProcessException()
+
+        args = self.map_arguments(match.groups())
+        func = getattr(target, self.member)
+
+        return self.map_return_value(func(*args))
+
+
+class Var(BaseCommand):
+    def __init__(self, target_member, read_regex=None, write_regex=None, argument_mappings=None,
+                 return_mapping=lambda x: None if x is None else str(x), doc=None):
+        self.member = target_member
+
+        self.read_pattern = re.compile(read_regex) if read_regex is not None else None
+        if self.read_pattern.groups != 0:
+            raise RuntimeError(
+                'Command regex for reading member \'{}\' contains '
+                'arguments.'.format(target_member))
+
+        self.write_pattern = re.compile(write_regex) if write_regex is not None else None
+
+        if self.write_pattern.groups != 1:
+            raise RuntimeError(
+                'Command regex for writing member \'{}\' contains more '
+                'than one argument.'.format(target_member))
+
+        self.argument_mappings = argument_mappings
+        self.return_mapping = return_mapping
+        self.doc = doc
+
+    @property
+    def patterns(self):
+        patterns = []
+
+        if self.read_pattern is not None:
+            patterns.append(self.read_pattern)
+
+        if self.write_pattern is not None:
+            patterns.append(self.write_pattern)
+
+        return patterns
```

Comment:
I'd never heard of `filter`, etc being deprecated or slated for removal. Searching for it led me to some really old (2003-2005) discussions revolving around a comment PEP 279. My feeling is that if it hasn't happened 13 years later, it's pretty safe to assume there is too much opposition to removing them and that they will stay.

I like `filter` here, especially the `None` variant. It's concise and clear, much more so than the list comprehension, which requires more... comprehending to understand. My vote is to use `filter` and prepare the pitchforks and torches in case some faceless tyrant tries to take it away from us.